frontend/src/pages/LogsPage/useGetLogs.ts

```diff
@@ -106,19 +106,41 @@ export const useGetLogs = ({
 		}, []),
 	})
 
-	const logTraceIds =
-		data?.logs.edges.reduce((acc, e) => {
-			if (e.node.traceID) {
-				acc.push(e.node.traceID)
+	const logTraceIdSet = new Set()
+	let latestLogTime, earliestLogTime
+
+	if (data?.logs.edges.length) {
+		for (const edge of data.logs.edges) {
+			if (edge.node.traceID) {
+				logTraceIdSet.add(edge.node.traceID)
+			}
+
+			if (!latestLogTime || latestLogTime < edge.node.timestamp) {
+				latestLogTime = edge.node.timestamp
 			}
-			return acc
-		}, [] as string[]) || []
+
+			if (!earliestLogTime || earliestLogTime > edge.node.timestamp) {
+				earliestLogTime = edge.node.timestamp
+			}
```

Comment:
Maybe over engineered - chose to iterate through rather than using first and last timestamp incase we want to support sorting by other attributes in future

Reply:
Consider using useMemo if this will be recalculated often.